popils.py

```diff
@@ -1,33 +1,47 @@
-
-import sys
 import pygame
-import random
+import argparse
 
 from popils_constants import *
 
+# read in 3SAT problem
+parser = argparse.ArgumentParser()
+input = parser.add_mutually_exclusive_group()
+input.add_argument('-i', '--instance', nargs='+', type=str,
+                   help="3SAT instance: (default = " + DEFAULT_3SAT + ")")
+input.add_argument('-f', '--filename',
+                   help="file containing an instance of 3SAT")
+args = parser.parse_args()
+
+# Currently these all just blindly accept the input they're given
+# TODO Could make input more robust against typos
+if args.instance:
+    three_sat = " ".join(args.instance)
+elif args.filename:
+    with open(args.filename) as file:
+        three_sat = file.readline()
+else:
+    three_sat = DEFAULT_3SAT
+
 # initialize all pygame submodules
 pygame.init()
 
 # window title
 pygame.display.set_caption('Test')
 
 clock = pygame.time.Clock()
-size = [500, 1000]
+window_size = [WINDOW_WIDTH, WINDOW_HEIGHT]
 block_size = 50  # block size = 50px TODO dynamic block sizing
-screen = pygame.display.set_mode(size)
-done = False
+# initialize drawing surface
+screen = pygame.display.set_mode(window_size)
 
 
+def place_sub_gadget(code, bottom_row, col):
+    for i in range(GADGET_HEIGHT):
```

Comment:
```suggestion
    for i in range(SUB_GADGET_HEIGHT):
```
recommend prepending SUB as the full gadget height would be 6 blocks